crates/shared/src/price_estimation/factory.rs

```diff
@@ -247,6 +250,38 @@ impl<'a> PriceEstimatorFactory<'a> {
         }
     }
 
+    fn create_native_estimator(
+        &self,
+        source: NativePriceEstimatorSource,
+    ) -> Result<(String, Arc<dyn NativePriceEstimating>)> {
+        match source {
+            NativePriceEstimatorSource::GenericPriceEstimator(kind) => {
+                let estimator = PriceEstimator {
+                    kind,
+                    address: Default::default(),
+                };
+                let instance = self.create_estimator(estimator)?.native;
+                Ok((
+                    estimator.name(),
+                    Arc::new(NativePriceEstimator::new(
+                        Arc::new(self.sanitized(instance)),
+                        self.network.native_token,
+                        self.native_token_price_estimation_amount()?,
+                    )),
+                ))
+            }
+            NativePriceEstimatorSource::OneInchSpotPriceApi => Ok((
+                "OneInchSpotPriceApi".into(),
+                Arc::new(OneInch::new(
```

Comment:
ubernit (feel free to ignore):

It might make sense to add in the `native` module:

```rust
pub use self::oneinch::OneInch;
```

And then refer to it here as:
```suggestion
                Arc::new(native::OneInch::new(
```

So that we don't confuse `OneInch` with its potential other meanings in this context (i.e. the 1Inch price estimator).

That being said, if we plan on removing the 1Inch price estimator altogether in favour of the colocated solver, then it is fine as-is.